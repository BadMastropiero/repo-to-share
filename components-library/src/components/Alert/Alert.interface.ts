export interface AlertProps {
  title?: string;
  description?: string;
  acceptText?: string;
  rejectText?: string;
  noCloseButton?: boolean;
  onAccept?: () => void;
  onReject?: () => void;
  isOpen?: boolean;
  setIsOpen: (val: boolean) => void;
  backdropBgColor?: string;
  primary?: boolean;
}
