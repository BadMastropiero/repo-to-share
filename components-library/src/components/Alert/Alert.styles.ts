import styled, {css} from 'styled-components';
import {respondAbove} from '../../utils';
import {Button} from '../Button';
import {Modal} from '../Modal';

const StyledAlert = styled(Modal)<{primary?: boolean}>(
  ({
    theme: {
      colors: {primary: primaryColor, primaryBg, primary100, primary600},
    },
    primary: isPrimary,
  }) => css`
    border: 4px solid ${isPrimary ? primary600 : primary100};
    padding: 26px;
    border-radius: 10px;
    background: ${isPrimary ? primaryColor : primaryBg};
    box-shadow: 0px 10px 15px -5px rgba(0, 101, 166, 0.4);
    min-width: 150px;
    ${respondAbove('laptop', {
      maxWidth: '40%',
    })}
  `
);

export const AlertContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-items: flex-start;
  gap: 20px;
  word-break: break-word;
`;

export const AlertHeader = styled.div`
  display: flex;
  flex-direction: row;
  column-gap: 20px;
  justify-content: space-between;
  align-items: center;
`;

export const AlertCloseButton = styled(Button)(
  ({
    theme: {
      colors: {primaryLight, primaryBg},
    },
    primary: isPrimary,
  }) => css`
    border: unset;
    background: transparent;
    div:first-child {
      background: transparent;
      svg:first-child {
        fill: ${isPrimary ? primaryBg : primaryLight};
      }
    }

    :hover {
      div:first-child {
        background: transparent;
        svg:first-child {
          fill: ${primaryBg};
        }
      }
    }

    box-shadow: unset;
    flex-shrink: 0;
  `
);

export const AlertButtonContainer = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
  justify-items: flex-end;
  gap: 20px;
  align-items: center;
  background: unset;
`;

export default StyledAlert;
