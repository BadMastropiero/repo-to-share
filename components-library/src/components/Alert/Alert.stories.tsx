import {useEffect, useState} from 'react';
import {ComponentMeta, ComponentStory} from '@storybook/react';

import {Button} from '../Button';
import Alert from './Alert';
import {Typography} from '../Typography';

export default {
  title: 'Components/Alert',
  component: Alert,
} as ComponentMeta<typeof Alert>;

export const Base: ComponentStory<typeof Alert> = args => {
  const {
    isOpen: visible,
    acceptText,
    rejectText,
    backdropBgColor,
    title,
    description,
    noCloseButton,
    primary,
  } = args;

  const [isVisible, setIsVisible] = useState(visible);
  const [label, setLabel] = useState<string>('Interact with the Alert to see it results');

  useEffect(() => {
    setIsVisible(visible);
  }, [visible]);

  const launchAlert = () => {
    setIsVisible(true);
  };

  const handleAccept = () => {
    setIsVisible(false); // this must be done by the component
    setLabel('Alert accepted');
  };

  const handleReject = () => {
    setIsVisible(false);
    setLabel('Alert rejected');
  };

  return (
    <>
      <Button onClick={launchAlert} presetSize="small" text="Launch Alert" primary rounded />
      <Typography padding="10px">{label}</Typography>
      <Alert
        title={title}
        description={description}
        acceptText={acceptText}
        rejectText={rejectText}
        onAccept={handleAccept}
        onReject={handleReject}
        isOpen={isVisible}
        setIsOpen={setIsVisible} // expose this prop
        backdropBgColor={backdropBgColor}
        noCloseButton={noCloseButton}
        primary={primary}
      />
    </>
  );
};

Base.args = {
  title: 'Alert Title',
  description: 'Alert description: Lorem ipsum',
  isOpen: false,
  backdropBgColor: undefined,
  acceptText: undefined,
  rejectText: undefined,
  primary: undefined,
};

Base.parameters = {
  design: {
    type: 'figma',
    url: 'https://www.figma.com/file/epHXucNHjke2rES69iOsaS/TetraImpacts?node-id=2211%3A4381&t=ZEYHAGM50sAejo8B-4',
  },
};
