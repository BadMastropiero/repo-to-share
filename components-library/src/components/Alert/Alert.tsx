import React from 'react';
import {Button} from '../Button';

import {Close} from '../Icon';
import {Typography} from '../Typography';

// Interfaces
import {AlertProps} from './Alert.interface';

// Styles
import StyledAlert, {
  AlertButtonContainer,
  AlertCloseButton,
  AlertContainer,
  AlertHeader,
} from './Alert.styles';

const Alert: React.FC<AlertProps> = ({
  title = 'Message Type',
  description = 'Message description',
  acceptText = 'ACCEPT',
  rejectText = 'CANCEL',
  onAccept,
  onReject,
  noCloseButton = false,
  isOpen = false,
  setIsOpen,
  backdropBgColor,
  primary = false,
}) => (
  <StyledAlert
    primary={primary}
    isOpen={isOpen}
    className="ti-alert"
    noDismissOnEsc
    backdropBgColor={backdropBgColor}
    setIsOpen={setIsOpen}
  >
    <AlertContainer className="ti-alert-container">
      <AlertHeader className="ti-alert-header">
        <Typography
          fontWeight="700"
          fontSize="16px"
          height="19px"
          lineHeight="19px"
          fontFamily="Roboto Flex"
          fontStyle="normal"
          colorPreset={primary ? 'secondary' : 'primary'}
          padding="0px 0px 0px 1px" // * Needed to correct typography left border shadowing first letter
        >
          {title}
        </Typography>
        {noCloseButton || (
          <AlertCloseButton
            className="ti-alert-close-btn"
            presetSize="small"
            afterIcon={Close}
            rounded
            onClick={onReject}
            primary={primary}
          />
        )}
      </AlertHeader>
      <Typography
        fontWeight="500"
        fontSize="12px"
        lineHeight="20.4px"
        fontFamily="Roboto Flex"
        fontStyle="normal"
        colorPreset={primary ? 'secondary' : 'disable'}
      >
        {description}
      </Typography>
      <AlertButtonContainer className="ti-alert-buttons-container">
        <Button
          className="ti-alert-reject-btn"
          presetSize="small"
          text={rejectText}
          rounded
          onClick={onReject}
          primary={!primary}
          presetType="text"
        />
        <Button
          className="ti-alert-accept-btn"
          presetSize="small"
          text={acceptText}
          rounded
          onClick={onAccept}
          primary={primary}
        />
      </AlertButtonContainer>
    </AlertContainer>
  </StyledAlert>
);

export default Alert;
