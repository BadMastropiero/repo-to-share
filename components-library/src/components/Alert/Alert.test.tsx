import {render} from '@testing-library/react';
import {composeStories} from '@storybook/testing-react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom';
import * as stories from './Alert.stories';

const {Base} = composeStories(stories);

describe('Alert test', () => {
  test('should match snapshot', () => {
    const tree = render(<Base isOpen />);
    expect(tree).toMatchSnapshot();
  });

  test('should appear, then disappear when `CANCEL` button clicked', () => {
    const tree = render(<Base />);
    const item = tree.getByText('Launch Alert');

    userEvent.click(item);

    const ButtonInsideCard = tree.getByText('CANCEL').parentElement;
    expect(ButtonInsideCard).toBeInTheDocument();

    if (ButtonInsideCard) {
      userEvent.click(ButtonInsideCard);
      expect(ButtonInsideCard).not.toBeInTheDocument();
    }
  });

  test('should appear, then disappear when `ACCEPT` button clicked', () => {
    const tree = render(<Base />);
    const item = tree.getByText('Launch Alert');

    userEvent.click(item);

    const ButtonInsideCard = tree.getByText('ACCEPT').parentElement;
    expect(ButtonInsideCard).toBeInTheDocument();

    if (ButtonInsideCard) {
      userEvent.click(ButtonInsideCard);
      expect(ButtonInsideCard).not.toBeInTheDocument();
    }
  });
});
