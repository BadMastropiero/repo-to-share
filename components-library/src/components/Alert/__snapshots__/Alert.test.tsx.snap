// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert test should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="sc-bcXHqe sc-dkrFOg gkrZeY ti-button "
      >
        <p
          class="sc-bcXHqe sc-eDvSVe Hvoxq"
        >
          Launch Alert
        </p>
        <div
          class="sc-gKPRtg dOMrEj"
          color="#C5CFDF"
        />
      </button>
      <div
        class="sc-bcXHqe sc-eDvSVe dfLAUW"
      >
        Interact with the Alert to see it results
      </div>
      <dialog
        class="sc-pyfCe gSsiAW sc-jrcTuL iNtUSI ti-alert"
        role="dialog"
      >
        <div>
          <div
            class="sc-kDvujY eDsSLD ti-alert-container"
          >
            <div
              class="sc-ipEyDJ iNHvtD ti-alert-header"
            >
              <div
                class="sc-bcXHqe sc-eDvSVe hVGdoA"
                font-family="Roboto Flex"
                font-size="16px"
                font-style="normal"
                font-weight="700"
                height="19px"
              >
                Alert Title
              </div>
              <button
                class="sc-bcXHqe sc-dkrFOg iLwNZv ti-button sc-csuSiG gUmHdw ti-alert-close-btn"
              >
                <div
                  class="sc-gswNZR eYhtjX"
                >
                  <svg
                    class="sc-jSUZER hAlpLQ ticons-close undefined"
                    color="#0065A6"
                    height="16px"
                    role="img"
                    viewBox="0 0 24 24"
                    width="16px"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12.05 13.85L7.27495 18.625C7.00828 18.8917 6.69162 19.025 6.32495 19.025C5.95828 19.025 5.64162 18.8917 5.37495 18.625C5.10828 18.3584 4.97495 18.046 4.97495 17.688C4.97495 17.3294 5.10828 17.0167 5.37495 16.75L10.175 11.95L5.34995 7.15002C5.08328 6.88336 4.94995 6.57502 4.94995 6.22502C4.94995 5.87502 5.08328 5.56669 5.34995 5.30002C5.61662 5.03336 5.92895 4.90002 6.28695 4.90002C6.64562 4.90002 6.95828 5.03336 7.22495 5.30002L12 10.05L16.725 5.32502C16.9916 5.05836 17.3083 4.92502 17.675 4.92502C18.0416 4.92502 18.3583 5.05836 18.625 5.32502C18.8916 5.59169 19.025 5.91236 19.025 6.28702C19.025 6.66236 18.8916 6.98336 18.625 7.25002L13.875 12.025L18.65 16.8C18.9 17.05 19.025 17.3584 19.025 17.725C19.025 18.0917 18.9 18.4 18.65 18.65C18.4 18.9 18.096 19.025 17.738 19.025C17.3793 19.025 17.075 18.9 16.825 18.65L12.05 13.85Z"
                    />
                  </svg>
                </div>
                <div
                  class="sc-gKPRtg bJoWGa"
                  color="#FFFFFF"
                />
              </button>
            </div>
            <div
              class="sc-bcXHqe sc-eDvSVe jAKBWM"
              font-family="Roboto Flex"
              font-size="12px"
              font-style="normal"
              font-weight="500"
            >
              Alert description: Lorem ipsum
            </div>
            <div
              class="sc-eDWCr hgvGgx ti-alert-buttons-container"
            >
              <button
                class="sc-bcXHqe sc-dkrFOg sc-hLBbgP gkrZeY kpnHDr ti-button ti-alert-reject-btn"
              >
                <p
                  class="sc-bcXHqe sc-eDvSVe kEAGbr"
                >
                  CANCEL
                </p>
                <div
                  class="sc-gKPRtg dOMrEj"
                  color="#C5CFDF"
                />
              </button>
              <button
                class="sc-bcXHqe sc-dkrFOg iLwNZv ti-button ti-alert-accept-btn"
              >
                <p
                  class="sc-bcXHqe sc-eDvSVe kEAGbr"
                >
                  ACCEPT
                </p>
                <div
                  class="sc-gKPRtg bJoWGa"
                  color="#FFFFFF"
                />
              </button>
            </div>
          </div>
        </div>
      </dialog>
    </div>
  </body>,
  "container": <div>
    <button
      class="sc-bcXHqe sc-dkrFOg gkrZeY ti-button "
    >
      <p
        class="sc-bcXHqe sc-eDvSVe Hvoxq"
      >
        Launch Alert
      </p>
      <div
        class="sc-gKPRtg dOMrEj"
        color="#C5CFDF"
      />
    </button>
    <div
      class="sc-bcXHqe sc-eDvSVe dfLAUW"
    >
      Interact with the Alert to see it results
    </div>
    <dialog
      class="sc-pyfCe gSsiAW sc-jrcTuL iNtUSI ti-alert"
      role="dialog"
    >
      <div>
        <div
          class="sc-kDvujY eDsSLD ti-alert-container"
        >
          <div
            class="sc-ipEyDJ iNHvtD ti-alert-header"
          >
            <div
              class="sc-bcXHqe sc-eDvSVe hVGdoA"
              font-family="Roboto Flex"
              font-size="16px"
              font-style="normal"
              font-weight="700"
              height="19px"
            >
              Alert Title
            </div>
            <button
              class="sc-bcXHqe sc-dkrFOg iLwNZv ti-button sc-csuSiG gUmHdw ti-alert-close-btn"
            >
              <div
                class="sc-gswNZR eYhtjX"
              >
                <svg
                  class="sc-jSUZER hAlpLQ ticons-close undefined"
                  color="#0065A6"
                  height="16px"
                  role="img"
                  viewBox="0 0 24 24"
                  width="16px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12.05 13.85L7.27495 18.625C7.00828 18.8917 6.69162 19.025 6.32495 19.025C5.95828 19.025 5.64162 18.8917 5.37495 18.625C5.10828 18.3584 4.97495 18.046 4.97495 17.688C4.97495 17.3294 5.10828 17.0167 5.37495 16.75L10.175 11.95L5.34995 7.15002C5.08328 6.88336 4.94995 6.57502 4.94995 6.22502C4.94995 5.87502 5.08328 5.56669 5.34995 5.30002C5.61662 5.03336 5.92895 4.90002 6.28695 4.90002C6.64562 4.90002 6.95828 5.03336 7.22495 5.30002L12 10.05L16.725 5.32502C16.9916 5.05836 17.3083 4.92502 17.675 4.92502C18.0416 4.92502 18.3583 5.05836 18.625 5.32502C18.8916 5.59169 19.025 5.91236 19.025 6.28702C19.025 6.66236 18.8916 6.98336 18.625 7.25002L13.875 12.025L18.65 16.8C18.9 17.05 19.025 17.3584 19.025 17.725C19.025 18.0917 18.9 18.4 18.65 18.65C18.4 18.9 18.096 19.025 17.738 19.025C17.3793 19.025 17.075 18.9 16.825 18.65L12.05 13.85Z"
                  />
                </svg>
              </div>
              <div
                class="sc-gKPRtg bJoWGa"
                color="#FFFFFF"
              />
            </button>
          </div>
          <div
            class="sc-bcXHqe sc-eDvSVe jAKBWM"
            font-family="Roboto Flex"
            font-size="12px"
            font-style="normal"
            font-weight="500"
          >
            Alert description: Lorem ipsum
          </div>
          <div
            class="sc-eDWCr hgvGgx ti-alert-buttons-container"
          >
            <button
              class="sc-bcXHqe sc-dkrFOg sc-hLBbgP gkrZeY kpnHDr ti-button ti-alert-reject-btn"
            >
              <p
                class="sc-bcXHqe sc-eDvSVe kEAGbr"
              >
                CANCEL
              </p>
              <div
                class="sc-gKPRtg dOMrEj"
                color="#C5CFDF"
              />
            </button>
            <button
              class="sc-bcXHqe sc-dkrFOg iLwNZv ti-button ti-alert-accept-btn"
            >
              <p
                class="sc-bcXHqe sc-eDvSVe kEAGbr"
              >
                ACCEPT
              </p>
              <div
                class="sc-gKPRtg bJoWGa"
                color="#FFFFFF"
              />
            </button>
          </div>
        </div>
      </div>
    </dialog>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
